### DISCUSSION TITLE
Insert a page number of a reference

### DISCUSSION THREAD

**Discussion by yakatz** [jgm/pandoc#9800] [Q&A] [answered]
I need to write something like:
```markdown
This issue is addressed later. See the reference to this topic on [page @page:my_reference].
```

At that other point, I have this content:
```markdown
blah blah [@thisCitation]{#page:my_reference}
```

I get a warning from citeproc that the citation is not found (obviously because it is not a section that can be cited).  
The generated LaTeX output does include a label for this phantomsection:
```tex
blah blah
\phantomsection\label{page:my_reference}{{[}\citeproc{ref-thisCitation}{12}{]}},
```
but the referencing location looks like this:
```tex
See the reference to this topic on {[}page
\citeproc{ref-page:my_reference}{\textbf{page:my_reference?}}{]}
```

Is there any way for me to get the page number of this fake section?

```
Version:
pandoc.exe 3.2
Features: +server +lua
Scripting engine: Lua 5.4

My execution: pandoc --lua-filter=tools/include-files.lua --lua-filter=tools/alignment.lua --filter=mermaid-filter.cmd --filter=pandoc-crossref --citeproc --lua-filter=tools/short-captions.lua --standalone --pdf-engine=xelatex -N --template=template-umd-dissertation.tex --highlight-style=haddock -o paper.tex paper.md 
```

**Comment by badumont** [comment, marked as answer]
Pandoc does not have a mechanism to refer by page (it would not make sense for all formats). I wrote a filter that covers this use case, using a different syntax than yours. You can find it [here](https://bastien-dumont.onmypc.net/git/bdumont/pandoc-lua-filters/src/branch/master/text-crossrefs).

**Reply by yakatz** [reply]
Thank you. The link in your README to your crossrefenum implementation is broken, but I assume this is your implementation: https://bastien-dumont.onmypc.net/git/bdumont/crossrefenum

**Reply by yakatz** [reply]
I actually had to modify it because my `\label`s were being created twice, so I just commented out the section of the lua script that does that. I think I have everything working now.

**Comment by badumont** [comment]
Oh, thank you for reporting that, this comes from a rather recent change in the LaTeX writer I had not come across before!

The crossrefenum package is available on the CTAN and included in TeXLive since 2023: adding \usepackage{crossrefenum} in your template should be enough.
